mne/cov.py

```diff
@@ -687,16 +695,31 @@ def _unpack_epochs(epochs):
         epochs = _unpack_epochs(epochs)
     else:
         epochs = sum([_unpack_epochs(epoch) for epoch in epochs], [])
+    assert isinstance(epochs, list)
 
     # check for baseline correction
-    for epochs_t in epochs:
-        if epochs_t.baseline is None and epochs_t.info['highpass'] < 0.5 and \
-                keep_sample_mean:
-            warn('Epochs are not baseline corrected, covariance '
-                 'matrix may be inaccurate')
-
-    for epoch in epochs:
+    if any(epochs_t.baseline is None and epochs_t.info['highpass'] < 0.5 and
+           keep_sample_mean for epochs_t in epochs):
+        warn('Epochs are not baseline corrected, covariance '
+             'matrix may be inaccurate')
+
+    orig = epochs[0].info['dev_head_t']
+    if not isinstance(on_mismatch, string_types) or \
+            on_mismatch not in ['raise', 'warn', 'ignore']:
+        raise ValueError('on_mismatch must be "raise", "warn", or "ignore", '
+                         'got %s' % on_mismatch)
+    for ei, epoch in enumerate(epochs):
         epoch.info._check_consistency()
+        if (orig is None) ^ (epoch.info['dev_head_t'] is None) or not \
```

Comment:
https://docs.python.org/3/library/stdtypes.html#bitwise-operations-on-integer-types

But I'll change it to `!=` since others might also find that clearer